dev_guide/builds.adoc

```diff
@@ -17,6 +17,74 @@ runnable images to be used on OpenShift. There are three types of builds:
 - STI build
```

Comment:
also those need to be links or something...  "overview: there are 3 types of builds" and then saying nothing about them and moving on to buildconfig isn't very helpful.  They are defined over here:
http://docs.openshift.org/latest/architecture/core_objects/builds.html#build-strategies

(the fact that we have two places for build information is a bit of a challenge)